abixen-platform-core/src/test/java/com/abixen/platform/core/controller/DashboardControllerIT.groovy

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2010-present Abixen Systems. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.abixen.platform.core.controller
+
+import com.abixen.platform.core.AbstractPlatformIT
+import com.abixen.platform.core.application.dto.DashboardDto
+import com.abixen.platform.core.application.dto.DashboardModuleDto
+import com.abixen.platform.core.application.dto.LayoutDto
+import com.abixen.platform.core.application.dto.PageDto
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.context.embedded.LocalServerPort
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+
+class DashboardControllerIT extends AbstractPlatformIT {
+
+    @LocalServerPort
+    private int port
+
+    @Autowired
+    private TestRestTemplate template
+
+    void setup() {
+    }
+
+    void "should find DashboardDto"() {
+
+        given:
+        final Long pageId = 1L
+
+        when:
+        final ResponseEntity<DashboardDto> responseEntity = template
+                .withBasicAuth(ADMIN_USERNAME, ADMIN_PASSWORD)
+                .getForEntity("http://localhost:${port}/api/page-configurations/${pageId}", DashboardDto.class)
+
+        then:
+        responseEntity.getStatusCode() == HttpStatus.OK
+
+        final PageDto pageDto = responseEntity.getBody().getPage()
+        pageDto != null
+        pageDto.getId() == 1L
+        pageDto.getDescription() == "This is a sample test page"
+        pageDto.getTitle() == "Sample test"
+        pageDto.getIcon() == "fa fa-lock"
```

Comment:
It is very long then. Should be decomposed on method checkPageDto, checkLayoutDto etc

Reply:
Let's keep the verifications in one test's method.